### PR TITLE
Split into separate model and view

### PR DESCRIPTION
**Pull request by chris-steele**
Includes changes made in https://github.com/adaptlearning/adapt-contrib-slider/pull/133

Note: this is part 1 of the exercise of splitting this in model/view - this part is just about handling the immediate split, part 2 will be more of a refactor to have the view respond to changes in the model and any other improvements that can be identified.

Therefore when reviewing please do go ahead and point out any obvious issues or 'quick wins' but maybe save anything more than that until part 2.

List of outstanding accessibility PRs: https://github.com/adaptlearning/adapt_framework/issues/2206.

### CODE REVIEW

File: js/sliderModel.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
+define([
+	'core/js/adapt',
```
Comment (oliverfoster) [inline js/sliderModel.js]: Indentation throughout.

File: js/sliderModel.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                // Format number
+                if (answer) {
+                    items.push({value: i, selected: false, correct: (i == answer)});
+                } else {
+                    items.push({value: i, selected: false, correct: (i >= range._bottom && i <= range._top)});
+                }
+            }
+
+            this.set('_items', items);
+            this.set('_marginDir', (Adapt.config.get('_defaultDirection') === 'rtl' ? 'right' : 'left'));
```
Comment (oliverfoster) [inline js/sliderModel.js]: object notation:
```js
this.set({
  _items: items,
  _marginDir: (Adapt.config.get('_defaultDirection') === 'rtl' ? 'right' : 'left'))
});
```

File: js/sliderModel.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                    _userAnswer: undefined
+                });
+                return;
+            }
+
+            var items = this.get('_items');
+            var userAnswer = this.get('_userAnswer');
+            for (var i = 0, l = items.length; i < l; i++) {
+                var item = items[i];
+                if (item.value == userAnswer) {
```
Comment (oliverfoster) [inline js/sliderModel.js]: Triple equals always instead of double

File: js/sliderModel.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+            _.each(this.get('_items'), function(item) {
+                item.selected = false;
+            }, this);
+        },
+
+        isCorrect: function() {
+            var numberOfCorrectAnswers = 0;
+
+            _.each(this.get('_items'), function(item, index) {
+                if(item.selected && item.correct)  {
```
Comment (oliverfoster) [inline js/sliderModel.js]: spacing between if and brackets

File: js/sliderView.js [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+
+        // Used by question to enable the question during interactions
+        enableQuestion: function() {
+            this.setAllItemsEnabled();
+        },
+
+        setAllItemsEnabled: function() {
+            var isEnabled = this.model.get('_isEnabled');
+
+            if (this.$slider) {
```
Comment (oliverfoster) [inline js/sliderView.js]: Return early.
```js
	    if (!this.$slider) {
                this._deferEnable = true; // slider is not yet ready
                return;
            }

            if (!isEnabled) {
                this.$('.slider-widget').addClass('disabled');
                this.$slider.prop('disabled', true);
                this.$slider.rangeslider('update', true);
                return;
            }
                
            this.$('.slider-widget').removeClass('disabled');
            this.$slider.prop('disabled', false);
            this.$slider.rangeslider('update', true);
          
```

File: properties.schema [hunk 6]
```diff
@@ -165,7 +165,7 @@
           "help": "Correct top range answer on the scale -- ignored when 'Correct Answer' is set"
         }
       }
-    },
+    },
```
Comment (tomgreenfield) [inline properties.schema]: @moloko, could you revert this?

**Pull request merged**